### ISSUE TITLE
parsing json-ld version 1.1 fails, when type is a literal

### ISSUE THREAD

**Issue by WhiteGobo** [RDFLib/rdflib#2747]
So im not sure if this is a bug. So i tested if there are any problems bumping up the default  version of json-ld to `1.1` and the roundtrip test failed. Background is #2606 

```json
[
  {
    "@id": "file:///somewhere/",
    "@type": [
      {
        "@value": "thing too"
      }
    ]
  },
  {
    "@id": "file:///somewhere/#",
    "@type": [
      {
        "@value": "another thing"
      }
    ]
  }
]
```
Cant pe parsed with version `1.1` but it works with `1.0`.

```python
Graph().parse(jsonfile, format="json-ld", version=1.1)
```

It seems that it tries to get an existing reference to the type `{"@value": "another thing"}` which is a dict and cant be hashed. It doesnt try to do this in version `1.0`.

**Comment by WhiteGobo** [comment]
Ok i'm pretty sure this is a bug in the serializer and not in the parser. `@type` should be a [term, a compact iri or a reference](https://www.w3.org/TR/json-ld11/#terms) and not a list/set of node or a value object.

I would change the parser to give more leniency and let it fallback to its behaviour of version `1.0`, till the bug in the serializer is repaired. We could remove this behaviour or at least raise a warning in the next major release of rdflib.

**Comment by edmondchuc** [comment]
> As far as i understand it should be "`@type`": "`http://xmlns.com/foaf/0.1/Person`" instead, without the `[...]`

I guess the serialized result you got there is just the JSON-LD in the expanded form. If you don't want it as a list, you can either use compaction. Behaviour of `@type` here is relevant? https://www.w3.org/TR/json-ld11/#using-set-with-type.

Sorry, haven't looked too much into how this is related to your original issue though.

**Comment by WhiteGobo** [comment]
Mh your right. I still think value objects arent allowed in that array. so things like `{"@type": [{"@value": "thing too"}]}` shouldnt be allowed. (ive changed my comment above to reflect this)

**Comment by edmondchuc** [comment]
Yeah, probably the JSON-LD processor in RDFLib is just not as strict. Your provided example fails in https://json-ld.org/playground/ when the value of `@type` is a [value object](https://www.w3.org/TR/json-ld11/#dfn-value-object).

JSON-LD example: https://tinyurl.com/2atu5gdy

**Comment by WhiteGobo** [comment]
Ive renamed the issue, because of the bug in the serializer. Now Roundtrip test with default jsonld version 1.1 will succeed.
